SPECIFICATION.md

```diff
@@ -0,0 +1,368 @@
+
+### Event Types
+
+#### generalAvailability
+*Category: Version Event*
+
+Indicates when a component version is released and available for use.
+
+**Additional Required Fields:**
+
+| Field | Type | Description |
+|-------|------|-------------|
+| `versions` | array[string] | Array of simple version strings (e.g., ["1.0.0", "1.1.0"]). Required for version-specific events. |
+| `ranges` | array[string] | Array of VERS-formatted version ranges (e.g., ["vers:npm/>=1.0.0|<2.0.0"]). Required for range-based events. |
+
+Example:
+```json
+{
+  "type": "generalAvailability",
+  "effective": "2023-01-01T00:00:00Z",
+  "versions": ["1.0.0"],
+  "ranges": [],
+  "published": "2023-01-01T00:00:00Z",
+  "modified": "2023-01-01T00:00:00Z",
+  "references": [
+    "https://example.com/release-notes"
+  ]
+}
+```
+
+#### endOfDevelopment
+*Category: Version Event*
+
+Indicates when the manufacturer ceases development of a component or service.
+
+**Additional Required Fields:**
+
+| Field | Type | Description |
+|-------|------|-------------|
+| `versions` | array[string] | Array of simple version strings (e.g., ["1.0.0", "1.1.0"]). Required for version-specific events. |
+| `ranges` | array[string] | Array of VERS-formatted version ranges (e.g., ["vers:npm/>=1.0.0|<2.0.0"]). Required for range-based events. |
+
+Example:
+```json
+{
+  "type": "endOfDevelopment",
+  "effective": "2024-01-01T00:00:00Z",
+  "ranges": ["vers:npm/>=1.0.0|<2.0.0"],
+  "versions": [],
+  "published": "2023-06-01T00:00:00Z",
+  "modified": "2023-06-01T00:00:00Z",
+}
+```
+
+#### endOfSupport
+*Category: Version Event*
+
+Indicates when the manufacturer ceases any and all support of a component or service. This point in time marks a transfer of risk from the manufacturer to the consuming organisation or user of the component or service, encompassing all cybersecurity knowledge and known vulnerabilities, with no further assistance provided by the manufacturer.
+
+**Additional Required Fields:**
+
+| Field | Type | Description |
+|-------|------|-------------|
+| `versions` | array[string] | Array of simple version strings (e.g., ["1.0.0", "1.1.0"]). Required for version-specific events. |
+| `ranges` | array[string] | Array of VERS-formatted version ranges (e.g., ["vers:npm/>=1.0.0|<2.0.0"]). Required for range-based events. |
+
+Example:
+```json
+{
+  "type": "endOfSupport",
+  "effective": "2024-01-01T00:00:00Z",
+  "versions": ["1.0.0", "1.1.0"],
+  "ranges": ["vers:npm/>=2.0.0|<3.0.0"],
+  "published": "2023-06-01T00:00:00Z",
+  "modified": "2023-06-01T00:00:00Z",
+}
+```
+
+#### endOfGuaranteedSupport
+*Category: Version Event*
+
+Indicates when the manufacturer no longer provides assured services such as technical assistance, user training, repairs, spare parts, and software updates for a component or service. Beyond this period, any support offered is discretionary and may incur extra costs or have restrictions.
```

Comment:
to me "guaranteed" implies some kind of contract, but open source doesn't have any guarantees - maybe there's a better term to use here?

Reply:
removed `endOfGuaranteedSupport` after concluding that claims like "guaranteed" can be represented in the support levels.